shuup/front/checkout/confirm.py

```diff
@@ -97,6 +100,29 @@ def get_context_data(self, **kwargs):
         context["product_ids"] = ','.join(self._get_product_ids())
         return context
 
+    def _get_guest_order_data_with_gdpr(self):
+        basket = self.basket
+        first_name = basket._billing_address.name
+        last_name = basket._billing_address.name_ext
+        email = basket._billing_address.email
+        basket._billing_address.save()
+        basket._shipping_address.save()
+        user, created = User.objects.get_or_create(
+            username=email,
+            defaults={
+                'email': email,
+                'first_name': first_name,
+                'last_name': last_name,
+                'is_active': False,
+            }
+        )
+        user.first_name = first_name
```

Comment:
Instead of this you could use `update_or_create`. In case your plan is to always update these.